### PR TITLE
#35106 - remove private note notifications from users without permission

### PR DESCRIPTION
**Pull request by Bonch2001**
This PR addresses an issue where users without the "view_private_notes" permission were receiving notifications for private notes. These users could neither view the private notes nor dismiss the associated notifications, leading to confusion and cluttered notification lists.

### CODE REVIEW

File: app/controllers/advanced_messenger_controller.rb [hunk 1]
```diff
@@ -148,14 +145,4 @@ def ignore_all_unread_entities(unread_entities, is_visible)
       format.js {render inline: "location.reload();" }
     end
   end
-
-  # private helper method
-  def is_journal_visible (journal)
```
Comment (daniela-mateescu) [inline app/controllers/advanced_messenger_controller.rb:153]: Aceasta metoda mai era folosita in acest controller astfel: 
image

Cred ca mutand-o in helper, nu mai este accesibila cu `:`

Cum acest caz este mai greu de testat pune te rog un log (cu `Rails.logger.debug`) in metoda de mai sus. Suspectez ca `:is_journal_visible` este `nil`. Daca da de vazut cum obtii referinta. Probabil direct cu `is_journal_visible`

Idem si pentru cealalta metoda

File: app/helpers/advanced_messenger_helper.rb [hunk 2]
```diff
@@ -61,14 +64,32 @@ def getUnreadNotificationsGroupByParentEntity(unread_notifications, getParent)
                 grouped_unread_notifications[parentEntity.id] = unread_notification_status
             end
             unread_notification_status.count += 1 
+            unread_notification_status.unread_journals << entity
         end
         return grouped_unread_notifications
     end
 
+    def entity_visible_to_user?(entity, project)
```
Comment (daniela-mateescu) [inline app/helpers/advanced_messenger_helper.rb:72]: Cred ca putem scapa de logica din aceasta metoda si sa delegam catre metodele de mai sus `is_journal_visible` si `is_message_visible`. Vad ca ce are in plus este user_can_view_issues. Poti adauga aceasta verificare si in `is_journal_visible`

File: app/models/unread_notifications_status.rb [hunk 3]
```diff
@@ -2,10 +2,12 @@ class UnreadNotificationsStatus
     attr_accessor :count
     attr_accessor :user
     attr_accessor :parent
+    attr_accessor :unread_journals
```
Comment (daniela-mateescu) [inline app/models/unread_notifications_status.rb:5]: Aceasta pare nefolosit. Este uitat? Daca da trebuie sters. 
Manual:  https://redmine.xops-online.com/issues/34798#ShTraining-Git-03-Pre-commit-inspection-review-each-change-Staging-view-and-index

File: app/helpers/advanced_messenger_helper.rb (hunk 2 above)
Comment (Bonch2001) [reply app/helpers/advanced_messenger_helper.rb:72]: Da, am mutat verificarea in `is_journal_visible` si acum o apelez pe ea in `getUnreadNotificationsGroupByParentEntity`.

File: app/models/unread_notifications_status.rb (hunk 3 above)
Comment (Bonch2001) [reply app/models/unread_notifications_status.rb:5]: Da, intradevar. L-am eliminat.

File: app/controllers/advanced_messenger_controller.rb (hunk 1 above)
Comment (daniela-mateescu) [reply app/controllers/advanced_messenger_controller.rb:153]: ok. E  bine

**Pull request merged**